source/common/upstream/load_balancer_impl.cc

```diff
@@ -833,6 +882,32 @@ HostConstSharedPtr EdfLoadBalancerBase::chooseHostOnce(LoadBalancerContext* cont
   }
 }
 
+double EdfLoadBalancerBase::applyAggressionFactor(double time_factor) {
+  if (aggression_ == 1.0 || time_factor == 1.0) {
+    return time_factor;
+  } else {
+    return std::pow(time_factor, 1.0 / aggression_);
+  }
+}
+
+double EdfLoadBalancerBase::applySlowStartFactor(double host_weight, const Host& host) {
+  auto host_create_duration = std::chrono::duration_cast<std::chrono::milliseconds>(
+      time_source_.monotonicTime() - host.creationTime());
+  if (host_create_duration < slow_start_window_ &&
+      host.health() == Upstream::Host::Health::Healthy) {
+    aggression_ = aggression_runtime_ != absl::nullopt ? aggression_runtime_.value().value() : 1.0;
+    aggression_ = std::max(0.0, aggression_);
```

Comment:
If the aggression somehow was set to be < 0, can you throw in a ENVOY_LOG_EVERY_POW_2 and log an error when this happens?

Reply:
Per above I think we can remove this error handling here?

Reply:
we should keep it, as we cannot do validation in PGV layer